crates/pallets/starknet/src/tests/re_execute_transactions.rs

```diff
@@ -41,22 +37,14 @@ fn re_execute_tx_ok() {
         assert_eq!(Starknet::contract_class_by_class_hash(erc20_class_hash), None);
         // All txs are there
         assert_eq!(res.len(), 5);
-
-        // Now let's check the TransactionInfos returned
-        let execution_info: Vec<(TransactionExecutionInfo, CommitmentStateDiff)> =
-            txs.iter().map(|tx| execute_transasction(tx.clone()).unwrap()).collect();
-        assert_eq!(res[0], execution_info[0]);
-        assert_eq!(res[1], execution_info[1]);
-        assert_eq!(res[2], execution_info[2]);
-        assert_eq!(res[3], execution_info[3]);
-        assert_eq!(res[4], execution_info[4]);
```

Comment:
We have a problem here, it's that we are never running the validation of the tx, only its execution. 
I'm not sure that this is what is expected of us.
This lead to nonce not being updated, and fees being different.

Reply:
It was the previous behavious, so I kept it. But I think it will need a change

Reply:
The problem with that is that the method used to execute the transaction for real is not the real code from the pallet. It is some custom implem, not extrinsic. So we are not testing that our re_execute produces the same output that the pallet. We are testing that the custom `execute_transasction` produces the same output as our re_execute.
Which has no security value as it is as likely for us to make an error in the custom `execute_transasction` than it is for our `re_execute` to contain one.

Reply:
Better remove it entrierly and think about an other way to check than having this fake sentiment of security with tests that don't actually tests what they are supposed to

Reply:
But the custom execute still does the bare minimum of calling the blockifier and getting the execution infos. Even if we make a mistake in either place, then at least something breaks and we can chck

Reply:
But most of it will just be a copy-paste of the `re_execute_transaction` content. It would be checking that the copy-pasted code does the same thing both times. That's not something we want to test.

Like: `re_execute_transaction` internally calls `Pallet::execute_transaction`, now let's write a `execute_transaction` that's internal calls `Pallet::execute_transaction`. Wow surprise, they both behave the same. That's not adding any security.

What we want to test for is the fact that `re_execute_transaction` produces the same output as sequentially executing the extrinsics. The thing you are describing does not fulfill this purpose, so no reason to add it.